Adding NyzulHelper addon.

GlarinAsura opened the pull request:
Initial commit of NyzulHelper addon.

GlarinAsura committed:
Adding NyzulHelper addon.

Initial commit of NyzulHelper addon.

z16 commented on addons/NyzulHelper/NyzulHelper.lua:
```diff
@@ -0,0 +1,306 @@
+--[[
+Copyright (c) 2020, Glarin of Asura
```
This should be `©`, not `(c)`. I notice you also use `PartyMaster` in a line below. Make sure you you use [this text](https://github.com/Windower/Lua/wiki/Writing-Addons#licensing) and replace the marked instances.

GlarinAsura replied:
Requested changes made.

z16 commented on addons/NyzulHelper/NyzulHelper.lua:
```diff
@@ -0,0 +1,306 @@
+windower.register_event('addon command', function(input, ...)
+
+    local cmd = string.lower(input)
+	local args = {...}
```
Use four spaces for indentation, everywhere. should be safe to do a blanket replace.

GlarinAsura replied:
Requested changes made.

z16 commented on addons/NyzulHelper/NyzulHelper.lua:
```diff
@@ -0,0 +1,306 @@
+-- Handle addon args
+windower.register_event('addon command', function(input, ...)
+
+    local cmd = string.lower(input)
```
This will error if the first parameter is not specified. The common idiom here is:
```lua
local cmd = input and input:lower() or 'help'
```
Or omit the last part, if you don't have/want a help text.

GlarinAsura replied:
Requested change made.

z16 reviewed (changes requested):
For the most part the addon looks fine, function-wise. Stylistically, since we try to go for a uniform style please use 4 spaces for one level of indentation, use single quotes for strings and add a new line at the end of the file, and prefer multi-line `if` over single-line.

Also, for the addon to be listed in the launcher you need to add an entry into the `addons.xml` file. You can copy the structure for other addons in there.

I wish I could provide some insight into the functional aspect as well, but I know too little about Nyzul to contribute, so I'll trust your judgment here.

GlarinAsura committed:
Made requested code and styling changes.

Fixed incorrect text in disclaimer.
Made indentation uniform.
Switched all string literals to use single qoutes.
Made suggested help command change.

GlarinAsura committed:
Added enty to addons.xml

GlarinAsura commented:
Thank you so much for taking the time to review my code.  I have made all of the requested changes.

GlarinAsura committed:
Fixed incorrect links,
